Inc/HALAL/Services/Communication/SPI/SPI.hpp

```diff
@@ -8,24 +8,37 @@
 #pragma once
 
 #include "PinModel/Pin.hpp"
-#include "Packets/RawPacket.hpp"
+#include "Packets/SPIPacket.hpp"
 #include "DigitalOutputService/DigitalOutputService.hpp"
+#include "DMA/DMA.hpp"
 #include "ErrorHandler/ErrorHandler.hpp"
 
 #ifdef HAL_SPI_MODULE_ENABLED
 
+#define MASTER_SPI_CHECK_DELAY 100000 //how often the master should check if the slave is ready, in nanoseconds
+
 //TODO: Hay que hacer el Chip select funcione a traves de un GPIO en vez de a traves del periferico.
 
 /**
  * @brief SPI service class. Abstracts the use of the SPI service of the HAL library.
  * 
  */
 class SPI{
-private:
+public:
+	/**
+	 * @brief Possible states of the SPI class
+	 */
+    enum SPIstate{
+    	IDLE = 0,
+		STARTING_PACKET,
+    	WAITING_PACKET,
+		PROCESSING_PACKET,
+    };
+
     /**
-     * @brief Struct which defines all data refering to SPI peripherals. It is
-     *        declared private in order to prevent unwanted use. Only 
-     *        predefined instances should be used.
+     * @brief Struct which defines all data referring to SPI peripherals. DO NOT MODIFY ON RUN TIME.
+     * declared public so HAL callbacks can directly access the instance
+     * NOT intended to be modified after the inscribe of the spi, configuration should be made in Runes.hpp
```

Comment:
maybe  consider declaring it as private? so it is properly encapsulated?

Reply:
HAL callbacks need to access the instance for multiple operations, which are not part of the class (they are overriden functions of the HAL), and they are run in an interruption two times for each packet, so i would prefer to add the less layers possible there. 

Also a user could be interested in modifying it _before_ the inscribe on the spi (which is not against its use) so he can have only one runes for two different boards, which change only the configuration from slave to master by using a flag.